tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationRuntimeTest.java

```diff
@@ -174,4 +179,25 @@ public void testAnnotAvailabilty() {
         Assert.assertEquals(mapValue.get(StringUtils.fromString("q")).toString(), "hello");
         Assert.assertEquals(mapValue.get(StringUtils.fromString("r")).toString(), "world");
     }
+
+    @Test(expectedExceptions = BLangRuntimeException.class,
+          expectedExceptionsMessageRegExp = "^error: \\{.*\\}InvalidUpdate message=Invalid update " +
+                  "of record field: modification not allowed on readonly value.*")
+    public void testRecordTypeAnnotationReadonlyValueEdit() {
+        BRunUtil.invoke(resultAccessNegative, "testRecordTypeAnnotationReadonlyValueEdit");
+    }
+
+    @Test(expectedExceptions = BLangRuntimeException.class,
+          expectedExceptionsMessageRegExp = "^error: \\{.*\\}InvalidUpdate message=Invalid update " +
+                  "of record field: modification not allowed on readonly value.*")
+    public void testAnnotationOnObjectTypeReadonlyValueEdit() {
+        BRunUtil.invoke(resultAccessNegative, "testAnnotationOnObjectTypeReadonlyValueEdit");
+    }
+
+    @Test(expectedExceptions = BLangRuntimeException.class,
+          expectedExceptionsMessageRegExp = "^error: \\{.*\\}InvalidUpdate message=Invalid update " +
```

Comment:
We can assert these also in Ballerina itself, we can maybe change these later.

See https://github.com/ballerina-platform/ballerina-lang/blob/02b2ec6c18ce25569748d7f904055a2ab03f85a4/tests/jballerina-unit-test/src/test/resources/test-src/expressions/mappingconstructor/spread_op_field.bal#L149

Reply:
shall we do this later after error ctor changes as well?

Reply:
Please create an issue for this too?